src/envoy/http/mixer/control.cc

```diff
@@ -45,9 +45,21 @@ Control::Control(const Config& config, Upstream::ClusterManager& cm,
 
 Utils::CheckTransport::Func Control::GetCheckTransport(
     Tracing::Span& parent_span) {
-  return Utils::CheckTransport::GetFunc(
-      *check_client_factory_, parent_span,
-      config_.config_pb().transport().attributes_for_mixer_proxy());
+  std::string serialized_forward_attributes;
+
+  if (!config_.config_pb()
+           .transport()
+           .attributes_for_mixer_proxy()
+           .attributes()
+           .empty()) {
+    config_.config_pb()
+        .transport()
+        .attributes_for_mixer_proxy()
+        .SerializeToString(&serialized_forward_attributes);
+  }
```

Comment:
This is bad for performance.  It is done per request, per Check call.  Could you do following
1) similar to check_client_factory and report_client_factory define here https://github.com/istio/proxy/blob/master/src/envoy/http/mixer/control.h#L54

add the serialized_forward_attributes string at Control class.

2) then just pass its reference to all other paces.

Reply:
Done